ci/ssl.sh

```diff
@@ -2,19 +2,22 @@
 
 set -eux
 
+# TEST_RUBY_MYSQL2_SSL_DIR: custom SSL directory.
+SSL_DIR=${TEST_RUBY_MYSQL2_SSL_DIR:-/etc/mysql}
+
 # Make sure there is an /etc/mysql
-mkdir -p /etc/mysql
+mkdir -p "${SSL_DIR}"
 
 # Copy the local certs to /etc/mysql
-cp spec/ssl/*pem /etc/mysql/
+cp spec/ssl/*pem "${SSL_DIR}"
 
 # Wherever MySQL configs live, go there (this is for cross-platform)
 cd $(my_print_defaults --help | grep my.cnf | xargs find 2>/dev/null | xargs dirname)
 
 # Put the configs into the server
 echo "
 [mysqld]
-ssl-ca=/etc/mysql/ca-cert.pem
-ssl-cert=/etc/mysql/server-cert.pem
-ssl-key=/etc/mysql/server-key.pem
+ssl-ca=${SSL_DIR}/ca-cert.pem
+ssl-cert=${SSL_DIR}/server-cert.pem
+ssl-key=${SSL_DIR}/server-key.pem
```

Comment:
As a note, we are applying the patch below in the mysql2 gem RPM package in the Fedora project. We are using the *.pem files in the original repository directory without copying.

```
diff --git a/spec/mysql2/client_spec.rb b/spec/mysql2/client_spec.rb
index 5861882..3f5cda8 100644
--- a/spec/mysql2/client_spec.rb
+++ b/spec/mysql2/client_spec.rb
@@ -153,9 +153,9 @@ RSpec.describe Mysql2::Client do # rubocop:disable Metrics/B
lockLength
     let(:option_overrides) do
       {
         'host'     => 'mysql2gem.example.com', # must match the certificates
-        :sslkey    => '/etc/mysql/client-key.pem',
-        :sslcert   => '/etc/mysql/client-cert.pem',
-        :sslca     => '/etc/mysql/ca-cert.pem',
+        :sslkey    => 'spec/ssl/client-key.pem',
+        :sslcert   => 'spec/ssl/client-cert.pem',
+        :sslca     => 'spec/ssl/ca-cert.pem',
         :sslcipher => 'DHE-RSA-AES256-SHA',
         :sslverify => true,
       }
---
2.38.1
```